core/templates/domain/topics_and_skills_dashboard/TopicsAndSkillsDashboardFilterObjectFactory.ts

```diff
@@ -0,0 +1,66 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating filters of frontend topics
+ * and skill dashboard
+ */
+
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+
+export class TopicsAndSkillsDashboardFilter {
+  category: string;
+  keyword: string;
+  sort: string;
+  status: string;
+  /**
+   * @param {String} category - category to filter
+   * @param {String} keyword - keyword to filter
+   * @param {String} sort - sort by filter value
+   * @param {String} status - status to filter
+   */
+  constructor(category, keyword, sort, status) {
+    this.category = category;
+    this.keyword = keyword;
+    this.sort = sort;
+    this.status = status;
+  }
+  /**
+   * Resets the filter object values
+   */
+  reset(): void {
+    this.category = '';
+    this.keyword = '';
+    this.sort = '';
```

Comment:
Shouldn't this and status be null rather than empty string (if they are enums)?

Reply:
Yes, linked these to the constants. Fixed!